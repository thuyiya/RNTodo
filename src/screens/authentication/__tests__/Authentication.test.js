import React from "react";
import renderer from "react-test-renderer";
import Auth from "../";

describe("<Auth />", () => {
  it("has 4 children", () => {
    const tree = renderer.create(<Auth />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it("Main container shoud coverd with SafeAreaView", () => {
    const tree = renderer.create(<Auth />).toJSON();
    expect(tree.type).toBe('RCTSafeAreaView');
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Auth />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
