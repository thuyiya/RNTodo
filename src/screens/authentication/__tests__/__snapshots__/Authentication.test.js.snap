// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": "auto",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#333333",
            "fontSize": 16,
            "fontWeight": "300",
            "margin": 8,
            "textAlign": "center",
          },
          {},
        ]
      }
    >
      No face or fingerprint scanner is available on the device.
    </Text>
    <Text
      style={
        [
          {
            "color": "#333333",
            "fontSize": 16,
            "fontWeight": "500",
            "margin": 0,
            "textAlign": "center",
          },
          {},
        ]
      }
    >
      Set Authentication to Proceed
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignContent": "center",
          "backgroundColor": "#0145a6",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "margin": 8,
          "marginBottom": 60,
          "opacity": 1,
          "padding": 8,
          "width": 160,
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontWeight": "500",
            "textAlign": "center",
          }
        }
      >
        Go to Settings
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
